coalib/bears/Bear.py

```diff
@@ -18,10 +22,27 @@
 from coalib.settings.FunctionMetadata import FunctionMetadata
 from coalib.settings.Section import Section
 from coalib.settings.ConfigurationGathering import get_config_directory
-
 from .meta import bearclass
 
 
+db = pdb.Pdb()
+db.prompt = colorit.color_front('(pdb)',50,200,50)
+def debug_mode_function(func,*args,**kwargs):
+    x = func(*args, **kwargs)
+    # The above line of code is also needed to call the debugger through
+    # decorators so that if any run() method is using decorators and also it
+    # is returning "sub_function()" and that sub_function() is
+    # yielding something.
+    results = []
+    print("start pdb")
+    try:
+        while True:
+            result = db.runcall(next, x)
+            results.append(result)
```

Comment:
Didn't we want to call `db.runcall(func, *args, **kwargs)` above outside the loop as well?

Reply:
Do you want me to step into decorators also?